### DISCUSSION TITLE
Zoom out shows the wrong percentage on the minimap.

### DISCUSSION THREAD

**Discussion by gintarasgraciovas** [clientIO/joint#2669] [Q&A] [answered]
### Introduction

I'm experiencing an issue related to zooming out in my app. The same issue can be observed in this demo:
[kitchensink](https://changelog.jointjs.com/rappid/examples/kitchensink/?_gl=1*1q2d1nu*_gcl_aw*R0NMLjE3MTY0NjY1NjAuRUFJYUlRb2JDaE1JazUtUW1PQ2poZ01WM1poUUJoMHRFZ3pSRUFBWUFpQUFFZ0tZa1BEX0J3RQ..*_gcl_au*MTAzOTUyMjQwMy4xNzEwMjY1ODkx*_ga*MTI4MTU2NDQ0Ny4xNzA1MDYxODYz*_ga_S3PN155Z28*MTcxNjQ2OTMzMS4xMDYuMS4xNzE2NDY5NDgxLjM4LjAuMA..)

When zooming out the graph, it sometimes shows duplicate zoom percentages such as 20%, 30%, or 40%. Is there any way to fix this?
![image](https://github.com/clientIO/joint/assets/146833191/2b0a5640-66ad-445e-9723-b6c70a035dab)
![image](https://github.com/clientIO/joint/assets/146833191/f706c6da-e5bc-4e44-91f9-4e497db64b6d)

### Steps to reproduce

1) Open Kitchensink demo project
2) Try to zoom in and zoom out graph
3) See that percentage value may be the same after two zoom operations performed

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
It happens because we use simply an `<input type="range"/>` with a `step` attribute set to `20` by default.

```ts
inputEl.step = 20;
inputEl.value = 99;
inputEl.value; // => '100'
```

The range can show `100` when the paper scale is set to `0.99`.

You can change the `step` to `1` to overcome this.

```ts
{
    type: 'zoom-slider',
    name: 'zoom-slider',
    group: 'zoom',
    step: 1
},
````

It will result in the user being able to slide to any integer zoom level though.
Other option is to round the zoom levels when pinching.

I can imagine a feature request to allow the slider follow given steps but to show the real paper scale value when set externally.

How the other tools deal with this?

**Reply by gintarasgraciovas** [reply]
works for me, thanks :)
